src/components/TradeWidget/TokenRow.tsx

```diff
@@ -48,22 +59,85 @@ const WalletDetail = styled.div`
   }
 `
 
+function renderOptionLabel(token: TokenDetails): React.ReactNode {
+  return (
+    <div
+      style={{
+        display: 'flex',
+        alignItems: 'center',
+      }}
+    >
+      <TokenImgWrapper
+        src={token.image}
+        alt={token.name}
+        style={{
+          margin: '0.25em 2em 0.25em 1em',
+        }}
+      />
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+        }}
+      >
+        <div>
+          <strong>{token.symbol}</strong>
+        </div>
+        <div>{token.name}</div>
+      </div>
+    </div>
+  )
+}
+
+function formatOptionLabel(
+  options: { token: TokenDetails },
+  labelMeta: { context: FormatOptionLabelContext },
+): React.ReactNode {
+  const { token } = options
+  const { context } = labelMeta
+  return context === 'value' ? (
+    <div>
+      <strong>{token.symbol}</strong>
+    </div>
+  ) : (
+    renderOptionLabel(token)
+  )
+}
+
+const customSelectStyles = {
+  control: (provided: CSSProperties): CSSProperties => ({ ...provided, border: 'none' }),
+  menu: (provided: CSSProperties): CSSProperties => ({ ...provided, minWidth: '300px' }),
+  valueContainer: (provided: CSSProperties): CSSProperties => ({ ...provided, minWidth: '4.5em' }),
+}
+
 interface Props {
-  tokenDetails: TokenDetails
+  token: TokenDetails
+  tokens: TokenDetails[]
   selectLabel: string
+  onSelectChange: (selected: TokenDetails) => void
 }
 
-const TokenRow: React.FC<Props> = ({ selectLabel, tokenDetails: { name, image } }: Props) => {
-  const selectId = selectLabel.replace(/[^a-zA-Z]/g, '')
+const TokenRow: React.FC<Props> = ({ token, tokens, selectLabel, onSelectChange }: Props) => {
+  const options = useMemo(() => tokens.map(token => ({ token, value: token.symbol, label: token.name })), [tokens])
 
   return (
     <Wrapper>
-      <TokenImgWrapper alt={name} src={image} />
+      <TokenImgWrapper alt={token.name} src={token.image} />
       <SelectBox>
-        <label htmlFor={selectId}>{selectLabel}</label>
-        <select name="tokenSelector" id={selectId}>
-          <option value="DAI">DAI</option>
-        </select>
+        <label>{selectLabel}</label>
+        <Select
```

Comment:
I think this Select will be used in more parts of the App. It wouldn't be a bad idea to create a component out of it

Reply:
Good idea, I can have a simple PR just for that.

Reply:
Will address on #148